File: ingestion/job-ingestion/app/models/job.py
from pydantic import BaseModel, ConfigDict, Field


class Job(BaseModel):

    model_config = ConfigDict(
        populate_by_name=True
    )

    external_id: str = Field(alias="externalId")
    title: str
    company: str
    location: str
    description: str
    job_url: str = Field(alias="jobUrl")
    source: str

File: backend/job-intelligence-backend/src/main/java/com/jobintelligence/platform/entity/Job.java
package com.jobintelligence.platform.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "jobs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Job {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private String company;

    private String location;

    private String description;

    @Column(name = "job_url")
    private String jobUrl;

    private String source;

    @Column(name = "external_id")
    private String externalId;
}